app/src/main/java/com/google/samples/apps/sunflower/data/UnsplashPagingSource.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.data
+
+import androidx.paging.PagingSource
+import com.google.samples.apps.sunflower.api.UnsplashService
+import retrofit2.HttpException
+import java.io.IOException
+
+private const val UNSPLASH_STARTING_PAGE_INDEX = 1
+
+class UnsplashPagingSource (
+    private val service: UnsplashService,
+    private val query: String
+) : PagingSource<Int, UnsplashPhoto>() {
+
+    override suspend fun load(params: LoadParams<Int>): LoadResult<Int, UnsplashPhoto> {
+        val page = params.key ?: UNSPLASH_STARTING_PAGE_INDEX
+        return try {
+            val response = service.searchPhotos(query, page, params.loadSize)
+            val photos = response.results
+            LoadResult.Page(
+                data = photos,
+                prevKey = if (page == UNSPLASH_STARTING_PAGE_INDEX) null else page - 1,
+                nextKey = if (page == response.totalPages) null else page + 1
+            )
+        } catch (exception: IOException) {
```

Comment:
Maybe catch `exception: Exception`? Otherwise it might throw if something else happens

Reply:
Yeah, that's safer